device/src/state_machines/session.rs

```diff
@@ -414,6 +414,7 @@ impl WaitingForRx {
                             if let Ok(PhyPayload::Data(DataPayload::Encrypted(encrypted_data))) =
                                 lorawan_parse(shared.radio.get_received_packet(), C::default())
                             {
+                                let confirmed = encrypted_data.is_confirmed();
```

Comment:
I failed to notice how this is going to be used. Did you mean `self.confirmed`?

Reply:
This was a mistake. Carry over from the confirmed uplink PR